### PR TITLE
Ignore erroneous auto-scale configuration, but warn

### LINKED ISSUE #9339: ERT fails in update step if AUTO_SCALE cannot match any observations
### What happened? (You can include a screenshot if it helps explain)

User report where there was a mismatch in the name of the observations and the string given to AUTO_SCALE, e.g. observations was named `obs_*` whereas the input for AUTO_SCALE was `OBS_*` . This caused ERT to fail with the following message:
EndEvent(failed=True, msg="cannot concat empty list\nopening handshake failed\nopening handshake failed\nNo observations active for group: ['OBS_*']")

### What did you expect to happen?

I would except ert to not crash in this case, and ideally we should give a warning that there are no matches during the config parsing to avoid typos and such?

### steps to reproduce

```Python
Add following line to poly.ert: ANALYSIS_SET_VAR OBSERVATIONS AUTO_SCALE OBS_*
run es_mda
```

### Environment where bug has been observed

- [ ] python 3.10
- [x] python 3.11
- [ ] python 3.12
- [ ] macosx
- [ ] rhel7
- [x] rhel8
- [ ] local queue
- [ ] lsf queue
- [ ] slurm queue
- [ ] openPBS queue

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #9339 

**Approach**
Detect and ignore (but warn)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [x] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by berland** [comment]
(test missing)

File: test-data/ert/poly_example/poly.ert [hunk 1]
```diff
@@ -6,9 +6,11 @@ RUNPATH poly_out/realization-<IENS>/iter-<ITER>
 OBS_CONFIG observations
 REALIZATION_MEMORY 50mb
 
-NUM_REALIZATIONS 100
+NUM_REALIZATIONS 10
```
Comment (eivindjahren) [inline test-data/ert/poly_example/poly.ert:9]: Not sure if we should change anything in the test-data. Maybe we could instead just edit this in the test.

**Review by eivindjahren — CHANGES_REQUESTED** [review]
Currently edits the test-data in place, this should instead just be handled in the test.

File: test-data/ert/poly_example/poly.ert (hunk 1 above)
Comment (berland) [reply test-data/ert/poly_example/poly.ert:9]: this is just a boo-boo, will fix. Note to self, remove  `git commit -a` from muscle memory.

Comment (eivindjahren) [reply test-data/ert/poly_example/poly.ert:9]: No problem, it was just in the ready for review column so thought I would comment on it to be certain.

Comment (berland) [reply test-data/ert/poly_example/poly.ert]: fixed

**Review by eivindjahren — APPROVED** [review]
:100:

**Pull request merged**